libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -417,6 +417,40 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPoseStamped& msg)
     }
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_Imu& msg)
+{
+    update_topic(msg.header.stamp);
+    strcpy(msg.header.frame_id, BASE_LINK_FRAME_ID);
+
+    auto &imu = AP::ins();
```

Comment:
If you want the IMU that the EKF is using. It needs to be defined how the lane switching works. 

Only the active IMU has notch filtering, so ROS might suddenly get noisy data.